### PR TITLE
Adds Azure Firewall

### PR DESCRIPTION
**Pull request by pabrus**
This PR is adding Azure Firewall to the reference hub-spoke implementation
Depends on https://github.com/mspnp/caf-secure-amp-infra/pull/14

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ Azure Onboarding is implemented using a single deployment script that wil create
 
 Use the following link to deploy it to a greenfield subscription:
 
-[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fcaf-secure-amp-infra%2Fpabrus%2Fhubspoke%2Fdeploy%2Fcaf-secure-deploy.json/createUIDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fcaf-secure-amp-infra%2Fpabrus%2Fhubspoke%2Fdeploy%2Fcaf-secure-ui.json)
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fcaf-secure-amp-infra%2Fpabrus%2Fazurefw%2Fdeploy%2Fcaf-secure-deploy.json/createUIDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fcaf-secure-amp-infra%2Fpabrus%2Fazurefw%2Fdeploy%2Fcaf-secure-ui.json)
```
Comment (pabrus) [inline README.md:65]: Reminder to revert this change

File: deploy/caf-secure-deploy.json [hunk 2]
```diff
@@ -215,10 +215,17 @@
...
+            "type": "array",
+            "metadata": {
+                "displayName": "Availability Zones for the Azure Firewall"
+            },
+            "defaultValue": []
         }
     },
     "variables": {
-        "templateRootUri": "https://raw.githubusercontent.com/mspnp/caf-secure-amp-infra/pabrus/hubspoke/deploy/auxiliary/",
+        "templateRootUri": "https://raw.githubusercontent.com/mspnp/caf-secure-amp-infra/pabrus/azurefw/deploy/auxiliary/",
```
Comment (pabrus) [inline deploy/caf-secure-deploy.json:228]: Reminder to revert this change

File: README.md (hunk 1 above)
Comment (ckittel) [reply README.md]: reverted

File: deploy/caf-secure-deploy.json (hunk 2 above)
Comment (ckittel) [reply deploy/caf-secure-deploy.json]: reverted

**Review by ckittel — APPROVED** [review]
I merged in the prior PR.  I didn't test the deployment, but looks good.  The only thing I would have done a bit different was adding a second IP address to the firewall.  Firewalls can run out of ports on a single IP quickly.  But that's really out of scope for this security-focused MVP.  When you're back (get better @pabrus!), you can make sure I didn't mess anything up in the merge :)

**Comment by pabrus** [comment]
@ckittel thanks for merging it.
